src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectSnapshotChangeTrigger.cs

```diff
@@ -11,11 +13,21 @@ internal class WorkspaceProjectSnapshotChangeTrigger : ProjectSnapshotChangeTrig
     {
         private ProjectSnapshotManagerBase _projectManager;
 
+        public int ProjectChangeDelay { get;  set; } = 3 * 1000;
```

Comment:
OK so go with me a sec. We need some kind of buffer on workspace changes because otherwise we do a literal ton of work trying to initialize state on the critical path for perf. This is just a simple temporal buffer, and it works pretty nicely form what I've been able to see.